pkgs/development/libraries/science/math/libamplsolver/default.nix

```diff
@@ -9,19 +9,23 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-LVmScuIvxmZzywPSBl9T9YcUBJP7UFAa3eWs9r4q3JM=";
   };
 
-  patches = [
-    # Debian provides a patch to build a shared library
-    (fetchpatch {
-      url = "https://sources.debian.org/data/main/liba/libamplsolver/0~20190702-2/debian/patches/fix-makefile-shared-lib.patch";
-      sha256 = "sha256-96qwj3fLugzbsfxguKMce13cUo7XGC4VUE7xKcJs42Y=";
-    })
-  ];
+  patches = if stdenv.isLinux
+  then
+    [
+      # Debian provides a patch to build a shared library
+      (fetchpatch {
+        url = "https://sources.debian.org/data/main/liba/libamplsolver/0~20190702-2/debian/patches/fix-makefile-shared-lib.patch";
+        sha256 = "sha256-96qwj3fLugzbsfxguKMce13cUo7XGC4VUE7xKcJs42Y=";
+      })
+    ]
+  else
+    [ ./libamplsolver-darwin.patch ];
 
   installPhase = ''
     runHook preInstall
     pushd sys.`uname -m`.`uname -s`
     install -D -m 0644 *.h -t $out/include
-    install -D -m 0644 *.so* -t $out/lib
+    install -D -m 0644 *.${if stdenv.isLinux then "so" else "dylib"}* -t $out/lib
```

Comment:
```suggestion
    install -D -m 0644 *.${stdenv.hostPlatform.extensions.sharedLibrary}* -t $out/lib
```

Reply:
fixed!

Reply:
Update: works, but needs to remove the . before `${stdenv.hostPlatform.extensions.sharedLibrary}*`